File: ueueo-framework/ueueo-domain-driven-desgin/src/main/java/com/ueueo/ddd/domain/entities/events/Distributed/EntityDeletedEto.cs
﻿using System;
using Volo.Abp.EventBus;
using Volo.Abp.MultiTenancy;

namespace Volo.Abp.Domain.Entities.Events.Distributed;

[Serializable]
[GenericEventName(Postfix = ".Deleted")]
public class EntityDeletedEto<TEntityEto> : IEventDataMayHaveTenantId
{
    public TEntityEto Entity;// { get; set; }

    public EntityDeletedEto(TEntityEto entity)
    {
        Entity = entity;
    }

    public   boolean IsMultiTenant(out ID tenantId)
    {
        if (Entity is IMultiTenant multiTenantEntity)
        {
            tenantId = multiTenantEntity.TenantId;
            return true;
        }

        tenantId = null;
        return false;
    }
}

File: ueueo-framework/ueueo-domain-driven-desgin/src/main/java/com/ueueo/ddd/domain/entities/events/distributed/EntityDeletedEto.java
package com.ueueo.ddd.domain.entities.events.distributed;

import com.ueueo.ID;
import com.ueueo.eventbus.GenericEventNameAttribute;
import com.ueueo.eventbus.IEventDataMayHaveTenantId;
import com.ueueo.multitenancy.IMultiTenant;
import lombok.Data;

@GenericEventNameAttribute(postfix = ".Deleted")
@Data
public class EntityDeletedEto<TEntityEto> implements IEventDataMayHaveTenantId {
    private TEntityEto entity;

    public EntityDeletedEto(TEntityEto entity) {
        this.entity = entity;
    }

    @Override
    public boolean isMultiTenant() {
        return entity instanceof IMultiTenant;
    }

    @Override
    public ID getTenantId() {
        if (entity instanceof IMultiTenant) {
            return ((IMultiTenant) entity).getTenantId();
        }
        return null;
    }

}
